addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -165,6 +170,42 @@ private void checkMetaElementsMatches(Source source) {
         }
     }
 
+    private void checkDomElementMatches(HttpMessage msg) {
+        Document doc = Jsoup.parse(msg.getResponseBody().toString());
+
+        for (Map<String, Map<String, Map<String, AppPattern>>> domSelectorMap :
+                currentApp.getDom()) {
+            for (Map.Entry<String, Map<String, Map<String, AppPattern>>> selectorMap :
+                    domSelectorMap.entrySet()) {
+                for (Map.Entry<String, Map<String, AppPattern>> nodeSelectorMap :
+                        selectorMap.getValue().entrySet()) {
+                    for (Map.Entry<String, AppPattern> value :
+                            nodeSelectorMap.getValue().entrySet()) {
+                        Elements selectedElements = doc.select(selectorMap.getKey());
+                        for (org.jsoup.nodes.Element selectedElement : selectedElements) {
+                            if (Objects.equals(value.getKey(), "text")) {
+                                AppPattern Ap = value.getValue();
+                                addIfMatches(Ap, selectedElement.text());
+                            }
+                            if (Objects.equals(nodeSelectorMap.getKey(), "attributes")) {
+                                AppPattern Ap = value.getValue();
+                                if (selectedElement.hasAttr(String.valueOf(value.getKey()))) {
+                                    addIfMatches(Ap, selectedElement.attr(value.getKey()));
+                                }
+                            }
+                            if (Objects.equals(nodeSelectorMap.getKey(), "properties")) {
```

Comment:
Could you add test for this case?

Reply:
There only one dom selector having properties in apps.json and I can't seem to find any example for it.

Reply:
Check tue upstream linked earlier for examples, or just make something up.

Reply:
Jsoup is not able to extract react-dom properties. Can you suggest a fix for that?
Maybe some other library.

Reply:
Could you provide an example?

Reply:
![Screenshot from 2020-12-31 21-01-07](https://user-images.githubusercontent.com/40137469/103416173-7b984680-4bab-11eb-9abc-4bb75e6e3f26.png)
Jsoup is unable to access "_reactRootContainer" property of "body > div"

Reply:
Can I use selenium for detecting property?